recipe-client-addon/lib/PreferenceExperiments.jsm

```diff
@@ -186,10 +187,39 @@ this.PreferenceExperiments = {
       lastSeen: new Date().toJSON(),
       preferenceName,
       preferenceValue,
+      preferenceType,
       previousPreferenceValue: preferences.get(preferenceName, undefined),
       preferenceBranchType,
     };
 
+    // If there's a previous preference value set, we need to ensure that the
+    // incoming value is of the same type.
+    if (preferences.has(preferenceName)) {
+      // get the previous type in PREF_ form
+      const previousType = Services.prefs.getPrefType(preferenceName);
+
+      let argumentValueType = preferenceType;
+      const typeConversions = {
+        boolean: Services.prefs.PREF_BOOL,
+        string: Services.prefs.PREF_STRING,
+        integer: Services.prefs.PREF_INT,
+      };
+
+      // Default to an INVALID pref type, unless there is a key in the typeConversions
+      // that matches the given argument type
+      let givenType = Services.prefs.PREF_INVALID || 0;
+      if(typeConversions.hasOwnProperty(argumentValueType)){
```

Comment:
Nit: spaces before `(` and after `)`

This should fail eslint (./mach eslint path/to/file inside m-c). Does it not?